apps/YellowSensorToCloud/components/sensors/light/lightSensor.c

```diff
@@ -16,34 +16,35 @@
 #include "interfaces.h"
 #include "periodicSensor.h"
 #include "lightSensor.h"
-#include "fileUtils.h"
+#include "iio.h"
 
-static const char LightFile[]  = "/driver/in_intensity_input";
+
+static psensor_Ref_t PeriodicSensorRef;
+static struct iio_context *LocalIioContext;
+struct iio_channel *IntensityChannel;
 
 
 //--------------------------------------------------------------------------------------------------
 /**
- * Read the light level measurement.
+ * Retrieve a light level sample from the sensor.
  *
  * @return LE_OK if successful.
  */
 //--------------------------------------------------------------------------------------------------
 le_result_t light_Read
 (
-    double* readingPtr
-        ///< [OUT] Where the reading (in W/m2) will be put if LE_OK is returned.
+    double* reading  ///< [OUT] Where the reading (in W/m2) will be put if LE_OK is returned.
 )
 {
     double light;
-    le_result_t r = file_ReadDouble(LightFile, &light);
-    if (r != LE_OK)
-    {
-        return r;
+    if (iio_channel_attr_read_double(IntensityChannel, "input", &light) != 0) {
+        LE_ERROR("Error when reading light sensor: %s", strerror(errno));
```

Comment:
`"%s", strerror(errno)` can be simplified to `"%m"`.

Reply:
done.  thanks